### ISSUE TITLE
Segmentation fault when the collector 

### ISSUE THREAD

**Issue by asutoshpalai** [census-instrumentation/opencensus-service#585]
If we start Collector at the same port as Agent, we get segmentation fault due to the way the error is handled at at [cmd/occollector/app/collector/receivers.go#L54](https://github.com/census-instrumentation/opencensus-service/blob/dee9655f45a480266d5ab80ef206b5faf97c2f04/cmd/occollector/app/collector/receivers.go#L54).

### Reproducing the error

Agent config
```yaml
receivers:
  opencensus:
    address: ":55678"

exporters:
  opencensus:
    endpoint: "localhost:55680"

zpages:
  port: 8884
```

Collector config
```yaml
collector-config.yaml 
log-level: DEBUG
receivers:
  opencensus:
    port: 55678

queued-exporters:
  jaeger-all-in-one:
    num-workers: 4
    queue-size: 100
    retry-on-failure: true
    sender-type: jaeger-thrift-http
    jaeger-thrift-http:
      collector-endpoint: http://localhost:14268/api/traces
      timeout: 5s

zpages-http-port: 9889
zpages:
  port: 8889
```

First start the agent and then the collector.

### Cause
```go
...
			rec, err := receiver.runFn(logger, v, traceConsumers, asyncErrorChan)
			if err != nil {
				...
				logger.Fatal("Cannot run receiver for "+rec.TraceSource(), zap.Error(err))
			}
...
```

If there is an error, `rec` might be `nil`, so calling `rec.TraceSource()` will give a segmentation fault.

**Comment by pjanotti** [comment]
Good find @asutoshpalai. Do you have cycles to submit a PR fixing it?

**Comment by asutoshpalai** [comment]
I am currently slammed at work, but I can definitely get around to it during the weekend!

**Comment by pjanotti** [comment]
Alright then! I'll assign this to you.

**Comment by asutoshpalai** [comment]
Fixed in #592
